### PR TITLE
contracts: make test work with debugger

### PR DESCRIPTION
**Pull request by pgherveou**
When debugging tests, the CARGO_* environment variables are not set, and thus the relative path fails to resolve.
using a path relative to PWD should make things work

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2624093

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2624094

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2624095

File: frame/contracts/src/tests.rs [hunk 1]
```diff
@@ -458,7 +458,7 @@ fn compile_module<T>(fixture_name: &str) -> wat::Result<(Vec<u8>, <T::Hashing as
 where
 	T: frame_system::Config,
 {
-	let fixture_path = ["fixtures/", fixture_name, ".wat"].concat();
```
Comment (juangirini) [inline frame/contracts/src/tests.rs:461]: Wouldn't this break the test when not in debug mode?

Comment (pgherveou) [reply frame/contracts/src/tests.rs:461]: tested locally I thought it was working, let see if the CI tests go ✅

Comment (pgherveou) [reply frame/contracts/src/tests.rs:461]: looks like I was wrong 🤔

File: frame/contracts/src/tests.rs [hunk 2]
```diff
@@ -458,7 +458,12 @@ fn compile_module<T>(fixture_name: &str) -> wat::Result<(Vec<u8>, <T::Hashing as
 where
 	T: frame_system::Config,
 {
-	let fixture_path = ["frame/contracts/fixtures/", fixture_name, ".wat"].concat();
+	let fixture_root = if std::env::var("CARGO_MANIFEST_DIR").is_ok() {
```
Comment (athei) [inline frame/contracts/src/tests.rs:461]: Shouldn't we use `CARGO_MANIFEST_DIR` in case it is set (making the path absolute)?

Comment (pgherveou) [reply frame/contracts/src/tests.rs:461]: that would be the default behavior happening behind the scene to resolve the absolute path, we could to it explicitly if that's what you mean

Comment (pgherveou) [reply frame/contracts/src/tests.rs:461]: ah I guess you are write we can getr the value and use it directly here, that's probably cleaner

Comment (athei) [reply frame/contracts/src/tests.rs:461]: Yes that is what I mean. Use the value instead of calling `is_ok`.

File: frame/contracts/src/tests.rs [hunk 3]
```diff
@@ -458,7 +458,14 @@ fn compile_module<T>(fixture_name: &str) -> wat::Result<(Vec<u8>, <T::Hashing as
 where
 	T: frame_system::Config,
 {
-	let fixture_path = ["fixtures/", fixture_name, ".wat"].concat();
+	let fixture_path = [
+		// When `CARGO_MANIFEST_DIR` is not set, rust resolve relative path from the root folder
```
Comment (pgherveou) [inline frame/contracts/src/tests.rs:462]:
```suggestion
		// When `CARGO_MANIFEST_DIR` is not set, rust resolves the relative path from the root folder
```

**Review by athei — APPROVED** [review]
Nice

**Review by agryaznov — APPROVED** [review]

**Comment by pgherveou** [comment]
bot merge

**Comment by athei** [comment]
Failure  unrelated to code change.

**Pull request merged**